pkg/analyzer/lib/file_system/physical_file_system.dart

```diff
@@ -62,7 +77,8 @@ class PhysicalResourceProvider implements ResourceProvider {
   final AbsolutePathContext absolutePathContext =
       new AbsolutePathContext(io.Platform.isWindows);
 
-  PhysicalResourceProvider(FileReadMode fileReadMode) {
+  PhysicalResourceProvider({String folder, FileReadMode fileReadMode})
```

Comment:
In general we cannot make `fileReadMode` a named parameter, because this is a breaking change. I'd like to remove it eventually, but let's not mix these two changes.

Also, the name `folder` is too generic. Let's name it `stateLocation` and use instead of `HOME + SERVER_DIR`.

Reply:
Done.